### DISCUSSION TITLE
Use the request state (pending?) on useFetch in the component

### DISCUSSION THREAD

**Discussion by ahoiroman** [nuxt/nuxt#17379] [Questions]
Hello everybody,

given the situation I got an method that should only get information from an API, if the function is actively triggered by the user, how would I handle the pending state?

```
<template>
    <div>
        <!-- pending is not defined here        -->
        <span v-if="pending">This is pending</span>
        <button @click="fetchData">Click me!</button>
    </div>
</template>
<script setup>

function fetchData() {
    const {pending, data: count} = useLazyAsyncData('profile', () => $fetch('/api/profile'))
}
</script>
```

Pending is not defined on page load - only when the function is triggered. How could I solve this issue, given the fact that there may be multiple pending states in this component, as multiple API-calls can be made and I'd like to reflect each pending state in the UI?

**Comment by Extarys** [comment]
I have the same issue. It's not definied because of the scope of the `const`. I can't seem to find a way to easily pass the information to the template.

EDIT: Found a partial answer here: https://github.com/nuxt/framework/discussions/3779

Still need to play with it and find out how to have a pending state for a loading message. :smile:

**Comment by Flanker72** [comment]
For this particular solution you may need to make your own pending/loading state like this:
```
<script setup lang="ts">
const pending = ref(false);
const count = ref(0);

async function fetchData() {
  pending.value = true;
  try {
    count.value = await $fetch("/api/profile");
  } catch (err) {
    console.dir(err);
  } finally {
    pending.value = false;
  }
}
</script>
```

Please, consider, that `useAsyncData` and `useLazyAsyncData` are designed to fetch data directly during component setup. They will return data that may be used within the template cause all variables defined within the `script setup` logic become available.
If you need to load data somewhere later - use your own approach.

**Reply by Extarys** [reply]
Yes I saw the warning at the top of the page that all those functions are design to be called directly in `setup`. I hope eventually they are compatible with functions or a new function that encapsulate those will be provided :/ but meanwhile your solution works great and is a bit cleaner that what I tried last night :) Thx
